### PR TITLE
Rewrite old testsuite

### PR DESCRIPTION
**Pull request by jreidinger**

### CODE REVIEW

File: timezone/test/Timezone_test.rb [hunk 1]
```diff
@@ -225,4 +228,317 @@
...
+          expect(subject.GetDateTime(true, false)).to eq "2020-02-05 - 08:13:57"
+        end
+      end
+
+      context "locale_format is set to true" do
+        it "returns stripped time output with locale format" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%c\""
+          ).and_return("stdout" => "St??5.????nor??2020,??08:18:27??CET\n")
```
Comment (mvidner) [inline timezone/test/Timezone_test.rb]: The ???s do not look right.

File: timezone/test/Timezone_test.rb [hunk 2]
```diff
@@ -225,4 +228,317 @@
...
+          expect(subject.GetDateTime(false, false)).to eq "2020-02-05 - 09:13:57"
+        end
+      end
+
+      context "locale_format is set to true" do
+        it "returns stripped time output with locale format" do
+          expect(Yast::SCR).to receive(:Execute).with(
+            path(".target.bash_output"),
+            "/bin/date \"+%c\" \"--date=now 3600sec\""
+          ).and_return("stdout" => "St??5.????nor??2020,??09:18:27??CET\n")
```
Comment (mvidner) [inline timezone/test/Timezone_test.rb]: More strange ?uestion ma?ks

Comment (jreidinger) [reply timezone/test/Timezone_test.rb]: well, it should be UTF8. Probably my locale is broken after testing yast2 CLI for language :)

File: language/test/Language_test.rb [hunk 3]
```diff
@@ -459,4 +455,159 @@
...
+      subject.language = "cs_CZ.utf8"
+      expect(subject.Export).to include("language" => "cs_CZ.utf8")
+    end
+
+    it "returns map with installed languages" do
+      subject.languages = "cs_CZ,en_US"
+      expect(subject.Export).to include("languages" => "cs_CZ,en_US")
+    end
+
+    it "returns map with use_uth8 if utf is not used" do
```
Comment (imobachgs) [inline language/test/Language_test.rb]: typo: `use_utf8`

File: language/test/Language_test.rb [hunk 4]
```diff
@@ -459,4 +455,159 @@
...
+      expect(subject.languages).to eq "de_DE,cs_CZ"
+    end
+
+    it "adds to installed languages language from map" do
+      subject.Import("languages" => "cs_CZ", "language" => "de_DE")
+
+      expect(subject.languages).to eq "cs_CZ,de_DE"
+    end
+
+    # TODO: Autoinst specific behavior
```
Comment (imobachgs) [inline language/test/Language_test.rb]: np: What about adding an empty "it" so we get the "pending" item in the results?

File: language/test/Language_test.rb [hunk 5]
```diff
@@ -459,4 +455,159 @@
...
+
+  describe "#Summary" do
+    it "returns string with html list" do
+      expect(subject.Summary).to be_a(::String)
+      expect(subject.Summary).to include("<ul>")
+    end
+  end
+
+  describe "#IncompleteTranslation" do
+    it "returns true if language translation is lower then threshold" do
```
Comment (imobachgs) [inline language/test/Language_test.rb]: typo: *than* threshold

File: language/test/Language_test.rb [hunk 6]
```diff
@@ -459,4 +455,159 @@
...
+    end
+
+    it "returns upper cased language if country part is missing" do
+      expect(subject.GetGivenLanguageCountry("de")).to eq "DE"
+    end
+  end
+
+  describe "#Read" do
+    context "really is set to true" do
+      it "reads language from localed.conf" do
```
Comment (imobachgs) [inline language/test/Language_test.rb]: typo: `locale.conf`

File: timezone/src/modules/Timezone.rb [hunk 7]
```diff
@@ -639,7 +639,7 @@ def GetDateTime(real_time, locale_format)
         ds = 0
         if @diff != 0
           out2 = Convert.to_map(
-            SCR.Execute(path(".target.bash_output"), "date +%z")
+            SCR.Execute(path(".target.bash_output"), "/bin/date +%z")
```
Comment (imobachgs) [inline timezone/src/modules/Timezone.rb]: I would use `/usr/bin/date` as `/bin/date` is a symlink to that location.

**Pull request merged**